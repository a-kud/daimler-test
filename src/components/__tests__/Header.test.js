import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from '../Header'

jest.mock('react-redux', () => ({
  ...require.requireActual('react-redux'),
  useDispatch: jest.fn()
}))

describe('Header Component', () => {
  const { useDispatch } = require('react-redux')
  useDispatch.mockImplementation(() => jest.fn())

  const wrapper = shallow(<Header />)
  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
