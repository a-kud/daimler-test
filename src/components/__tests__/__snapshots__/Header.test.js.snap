// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
<div
  className="makeStyles-grow-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))
      className="makeStyles-toobarRoot-7"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-2"
        noWrap={true}
        variant="h6"
      >
        Moovies for Jeff
      </WithStyles(ForwardRef(Typography))>
      <div
        className="makeStyles-search-3"
      >
        <div
          className="makeStyles-searchIcon-4"
        >
          <SearchIcon />
        </div>
        <WithStyles(ForwardRef(InputBase))
          classes={
            Object {
              "input": "makeStyles-inputInput-6",
              "root": "makeStyles-inputRoot-5",
            }
          }
          inputProps={
            Object {
              "aria-label": "search",
            }
          }
          onChange={[Function]}
          placeholder="Searchâ€¦"
        />
      </div>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
